File: lua/basics/fizzbuzz.lua
-- Print FizzBuzz for 1 through 100, choosing the label from the remainders.
-- Run: lua fizzbuzz.lua

local function label(n)
  if n % 15 == 0 then
    return "FizzBuzz"
  elseif n % 3 == 0 then
    return "Fizz"
  elseif n % 5 == 0 then
    return "Buzz"
  end
  return tostring(n)
end

for n = 1, 100 do
  print(label(n))
end

File: scala/basics/FizzBuzz.scala
// Print FizzBuzz for 1 through 100, choosing the label by matching the pair of remainders.
// Run: scala-cli run FizzBuzz.scala

def label(n: Int): String = (n % 3, n % 5) match
  case (0, 0) => "FizzBuzz"
  case (0, _) => "Fizz"
  case (_, 0) => "Buzz"
  case _      => n.toString

@main def fizzBuzz(): Unit =
  for n <- 1 to 100 do println(label(n))
